### ISSUE TITLE
Program doesn't save settings on reboot.

### ISSUE THREAD

**Issue by TrazzyDev** [ShareX/ShareX#702]
I've rebooted my computer twice while using the program after going through all the settings and getting it working how I want it to, but right after the reboot, everything is undone. Safe to say I won't be using this program until it's fixed because it takes so much time to fine tune all of the settings.

**Comment by Jaex** [comment]
When each setting windows close it saving settings. You can check debug log to see it yourself. So I don't know why it not saves for you.

**Comment by ku4a** [comment]
I think I know what's going on. I had described problem when set up the specific personal folder for ShareX in Application settings. Meanwhile in other utility I set up the same folder for automatic cleaning.  Therefore ShareX settings removed every time during the deleting files by the utility trigger. I think it is not obvious that setting "ShareX personal folder" affects to ShareX user upload settings.

**Comment by hmksq** [comment]
I could reproduce with ShareX 10.0! The program settings saves fine. But the dropdown list on the side of the application. Those settings don't save on reboot most of the time!

**Comment by Jaex** [comment]
People could reproduce it 8 years ago in ZScreen too. And still I have no idea why in few people Windows not waits for all applications to close before shutdown. Because it terminates ShareX before it can save settings I think.

**Comment by lnishan** [comment]
On Windows 10 Ent. 64bit and ShareX 10.1, this issue seems to happen everytime.

Can I manually modify the settings anywhere? (doesn't seem to be in %appdata% as most programs do these days)

**Comment by Jaex** [comment]
Many people reported here in Windows 10 upgrade their Documents/ShareX folder have ownership problems therefore ShareX couldn't save.

**Comment by lnishan** [comment]
Thank you for the response.

I had a fresh install of Windows 10 though. (64bit Enterprise, formatted my drive before installation)

**Comment by Jaex** [comment]
When you close ShareX manually is it saving settigs?

**Comment by lnishan** [comment]
No.
I also had process monitor (from sysinternals.com) turned on,
and I didn't see any relevant file access events either.

![procmon screenshot](http://i.imgur.com/ATItEuK.png)

**Comment by Jaex** [comment]
Can you check Debug -> Debug log is there any error.

**Comment by lnishan** [comment]
Sorry, but can you point me to it?
I don't know where the debug log is.

**Comment by Jaex** [comment]
![](https://dl.dropboxusercontent.com/u/14076298/ShareX/2015/08/ICxBAbv8k0.png)

**Comment by lnishan** [comment]
I suppose there's a switch somewhere? or?
![Tray Right Click Pic](http://i.imgur.com/brGG1wU.png)

**Comment by Jaex** [comment]
Seriously?

**Comment by lnishan** [comment]
Do I need to build ShareX from source?

**Comment by Jaex** [comment]
It is in main window.

**Comment by lnishan** [comment]
Oh I see... Now I really feel stupid.
I've actually never dbl-clicked the tray icon.

**Comment by lnishan** [comment]
Okay, so here's what I get.

No error messages to begin with.
1. When I set the settings through the right-click menu, nothing happens (ie. Nothing shows up in the log)
2. When I set the settings through the Main window, nothing happens **until** I close the main window.

The log
~~~~~
2015-08-24 06:32:55.418 - ShareX 10.1 started
2015-08-24 06:32:55.418 - Operating system: Microsoft Windows NT 10.0.10240.0
2015-08-24 06:32:55.418 - Command line: "C:\Program Files\ShareX\ShareX.exe" -silent
2015-08-24 06:32:55.418 - Personal path: C:\Users\lnishan\Documents\ShareX
2015-08-24 06:32:55.419 - Git: https://github.com/ShareX/ShareX/tree/ffb8d351394653e32418ed7e5555982158ee0426
.
.
.
2015-08-24 06:37:04.052 - ApplicationConfig save started: C:\Users\lnishan\Documents\ShareX\ApplicationConfig.json
2015-08-24 06:37:04.054 - HotkeysConfig save started: C:\Users\lnishan\Documents\ShareX\HotkeysConfig.json
2015-08-24 06:37:04.055 - UploadersConfig save started: C:\Users\lnishan\Documents\ShareX\UploadersConfig.json
2015-08-24 06:37:04.184 - UploadersConfig save successful: C:\Users\lnishan\Documents\ShareX\UploadersConfig.json
2015-08-24 06:37:04.189 - HotkeysConfig save successful: C:\Users\lnishan\Documents\ShareX\HotkeysConfig.json
2015-08-24 06:37:04.226 - ApplicationConfig save successful: C:\Users\lnishan\Documents\ShareX\ApplicationConfig.json
~~~~~

**Comment by lnishan** [comment]
So it might be a minor bug with the right-click context menu I guess.

**Comment by Jaex** [comment]
Both main window and tray menu using same function to open Settings windows and in both cases it saves settings when window close.

**Comment by lnishan** [comment]
The settings are saved only until Main window is closed.
Exiting tray menu does not trigger that.

**Comment by Jaex** [comment]
Exiting settings window trigger that.

**Comment by lnishan** [comment]
I'll do a screen recording. Hold on... (I'll edit this comment when completed)

**Comment by Jaex** [comment]
You don't need to do screen recording i know how saving works.

**Comment by lnishan** [comment]
Changing the settings through the tray menu doesn't leave any logs, nor does it change the config file.

I don't know the structure of ShareX. 
My guess is the Settings Saving function is probably binded to the wrong event (on main window close, instead of on setting window exit), given your description of it.

**Comment by Jaex** [comment]
Nothing is wrong...

It saves when Settings windows closed or Main window minimized to tray or ShareX itself closed. I already excessively saving unlike other softwares.

**Comment by lnishan** [comment]
>> It saves when Settings windows closed
Maybe you can check this part.

Because if I close settings page AND minimize Main window,
I'm seeing only 1 batch of logs
![image](https://cloud.githubusercontent.com/assets/6169473/9430752/c31a9aba-4a32-11e5-8347-0b4c51522d2f.png)

**Comment by Jaex** [comment]
I already checked it saves when I close window.

**Comment by lnishan** [comment]
Then it's really really weird I guess...

Thank you for taking time to look into it regardless...

**Comment by makedir** [comment]
Is this a joke? So this problem exists since half a year and still not fixed?

**Comment by zikeji** [comment]
@makedir this is a small problem that affects a small amount of people. One
that has proven difficult to replicate.

**Comment by makedir** [comment]
Give inport/export option, give option to select db/config folder, or put it under documents not in appdata.

**Comment by Jaex** [comment]
This problem exists for last 8 years and it is not fixable. Also I can't reproduce it too.

**Comment by makedir** [comment]
Cant reproduce?? I just installed it normally on a normal Windows 10
installation, and there it happened...

**Comment by Jaex** [comment]
I'm using ShareX for last 8 years and don't remember my settings reset.

**Comment by zikeji** [comment]
@makedir I've used ShareX on two desktops, one laptop, and one tablet (Atom processor) with WIndows 10 and had no issue, local accounts and a Microsoft account.

My best guess would be a permissions issue. Try opening ShareX as an administrator (close it first though), set your settings, close it from the taskbar (right click > quit) and then restart.

**Comment by Jaex** [comment]
My guess is Windows 10 shut downs too fast and not let ShareX complete its setting save. Dunno why this happens too. Normally it is supposed to wait for process to close.

**Comment by lnishan** [comment]
I would second @zikeji 's guess.
From what I observed, it does look like a permission issue. (if we rule out the possibility of bugs with source code)

**Comment by zikeji** [comment]
@Inishan From when I was writing an uploader for ShareX their logging / debugging backend is fairly catch-all. If any bug caused it'd it would have been fixed right now.

The destination settings window saves when you close that window, but the rest save when the application is closed completely. It may be worthwhile to add a button / menu item somewhere to force save the running settings or save them whenever a setting is changed.

**Comment by Jaex** [comment]
I was think about adding timer to save like every 10 minutes but then it won't be clean solution.

**Comment by makedir** [comment]
Why is there no save button or warning if closing window does not save
settings, like normal software does? As good as the program is and I really
like it, it has tons of bad human machine interaction designs. Or message
that settings were saved. Also no option for config folder. Or option to
put configs in local documents folder. Also I did start the program as
admin the first time which did actually mess things up in the first place I
assume not saving the configs with the correct user rights. And as an
offtopic... I just saw the "tools" like irc, ping, hash, twitter and so one
and I was... really?? Why does a screencap tool has included irc clients,
ping tester, dns tester, twitter client, and whatsoever... Was the dev
really that bored? Also option "index folder" is really bad named and no
one can know what it does until it is too late and automatically uploading
your private files so pastebin...

**Comment by zikeji** [comment]
>Why is there no save button or warning if closing window does not save
settings

The following menus save once you close their respective menu:
* Destination settings...
* Applications settings...
* Task settings...
* Hotkey settings...

So the majority of the settings do save when you need them to.

But the settings in these menus:
* After capture tasks
* After upload tasks
* Destinations
  * (excluding Destination settings...)

Do not save until you completely shut down ShareX. You can check the debug log to see when particular settings are saved and test for yourself.

> As good as the program is and I really like it, it has tons of bad human machine interaction designs.

This is superuser software, it's meant to be powerful. In fact, I would say the human-machine interaction is better than other software I've used for similar purposes. It's all laid out very plainly, capture captures, upload uploads, etc.

> Or message that settings were saved.

Click `Debug` and then `Debug Log` to see when it gets saved.

> Also no option for config folder. Or option to put configs in local documents folder.

Now I'm starting to think you're just trolling us. Config folder **by default** is in your Documents/ShareX folder, **AND** you can change that location for the hotkey settings and the uploaders settings (granted it's missing tasks and application settings).

> Also I did start the program as admin the first time which did actually mess things up in the first place I assume not saving the configs with the correct user rights.

Oh man that would have been nice to know a long while ago.

* Go to your documents folder, find the folder named "ShareX", delete it.
* Uninstall ShareX.
* Install ShareX.

See if that helps.

> And as an offtopic... I just saw the "tools" like irc, ping, hash, twitter and so one and I was... really?? Why does a screencap tool has included irc clients, ping tester, dns tester, twitter client, and whatsoever...

Major flaw in your logic there: **ShareX** is not "screencap software", it's sharing software. Link sharing, file sharing, text sharing, & image sharing. Those tools are just to allow powerusers more functionality. If you want to automatically share on your favorite IRC channel, automatically share to twitter, etc. I agree the DNS changer & ping tester are kind of random, but when you're dealing with uploading stuff they are relevant.

**Comment by Lunatrius** [comment]
> Oh man that would have been nice to know a long while ago.
>    * Go to your documents folder, find the folder named "ShareX", delete it.
>    * Uninstall ShareX.
>    * Install ShareX.

That won't fix it unless you terminate the process immediately after install.

Long story short: the ShareX installer runs ShareX as an administrator instead of the currently logged in user. Any (config) files that didn't exist before that will be owned by the administrator account, effectively "disabling" (config) saving.

**Comment by Jaex** [comment]
When task completed if there is no active task going on (if this is last active task) then ShareX will save all settings now. This way even if save fails in Windows shutdown still you will have up to date settings. I don't have like this frequent saves but I don't see any other solution if Windows not let ShareX to save settings while shutdown happens.

**Comment by dv2y** [comment]
Not sure if related, but the program fails to save its settings (any of them) if the settings file is for some reason hidden.

**Comment by ConcernedHobbit** [comment]
Just had this happen on Win10 after first time install. Had set custom URL shortening as well as FTP upload, really can't be arsed to re-do them since they were complicated.

**Comment by ConcernedHobbit** [comment]
Just checked other settings and can verify what @hmksq said about it being the dropdowns. Sadly, my custom FTP and bit.ly settings were also wiped. Seems to be Destinations doesn't get saved.

**Comment by Zacchino** [comment]
Same issue here. FTP settings are intact shortcut intact, but some other settings lost. Sorry I wish I had time to be more precise. Do you need a log or something to repro this bug?
It happens on Reboot of Windows 10 x64 1709.

**Comment by Lunatrius** [comment]
This will keep happening because the program runs with admin permissions after you install/update it.

Once you restart ShareX it will not be able to update any of the configs that were created as an admin, because you lack the permissions to do so.

This has to be fixed in the installer, to make sure that ShareX runs as the logged in user and not as an admin.

**Comment by artjos** [comment]
My path settings aren't saved, even as admin.

**Comment by Jaex** [comment]
#3668 fixes problem about not be able to save settings on Windows shutdown.

**Comment by tommynaranjo** [comment]
Sorry, what is the actual fix for this? I am confused.

**Comment by krusic22** [comment]
This issue can still happen. If you have the program set to start on startup, and you do a bunch of fast reboots/windows crashed, the settings will reset, despite you not even touching the program. 
Really annoying if you are doing a overclocking session and your settings keep resetting.

**Comment by BetTD** [comment]
> This issue can still happen. If you have the program set to start on startup, and you do a bunch of fast reboots/windows crashed, the settings will reset, despite you not even touching the program.
> Really annoying if you are doing a overclocking session and your settings keep resetting.

I was able to reproduce this issue on ShareX 13.1. I turned off my PC last night, but it decided to boot up on its own again, so I waited until Windows showed the login screen and then I shutdown my PC from there. Windows starts loading your session in the background even when you're not logged in, so I think ShareX managed to be loading up when I stopped the PC again.

This is very annoying because I had a bunch of custom uploaders and passwords that I did not have elsewhere.

**Comment by thomasddn** [comment]
I have to say that my settings aren't saved either. Stuff like hotkeys, theme (I prefer light), after capture tasks are reset after reboot. I tried to edit the settings file manually and even explicitly quit ShareX manually before reboot. Settings are restored when starting ShareX manually again, but after reboot settings are gone.

Debug log: 2020-04-01 09:03:42.035 - ApplicationConfig save successful: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
ShareX 13.1
Windows 10 (1809)

**Comment by dv2y** [comment]
I've noticed that when I set things up while running the program as administrator, the configurations I made stays after rebooting, otherwise is goes away.

**Comment by thomasddn** [comment]
@dv2y Tried that, but in the end my configuration was lost again.

I have to add that it not happens after every reboot, it just happens "all of the sudden". I configure the app to my needs and after a while it's gone again.

Additional notes:
- I'm not using hibernation, I shut down every day.
- It's a computer from work.
- I have admin rights.
- Antivirus is Kaspersky.

I hope that this long standing issue gets resolved somehow, because it's annoying to reconfigure again and again.

**Comment by Jaex** [comment]
> @Jaex , is it possible to reopen this issue? Clearly multiple people are having this issue.

Past 11 years I'm trying to figure out why peoples settings reset and added like 5 different protections to make sure in case of corruption in one action still setting can be kept, yet it still wiping. So, no I can't figure it out.

**Comment by Jaex** [comment]
Like even 6 hours ago I added one more protection as you can see from here: https://github.com/ShareX/ShareX/commit/14b9491664db82ffcce4207267fc7e446c594d26 which shows I still trying to figure it out all this time.

**Comment by thomasddn** [comment]
Thanks @Jaex! I wasn't sure this issue was still being worked on since it is marked as closed. Glad to hear you're still trying to fix it.

I'm happy to provide you with logs or things alike whenever I run into the problem. The thing is I don't use the app every day, but I'll keep a close eye to it for a while.

**Comment by thomasddn** [comment]
Yesterday before I shutdown my computer, I took a backup of the settings files to a completely different folder. Today I compared the backups with the files in the application folder and noticed that the application settings files are not json but binary or something.

Snippet of ApplicationConfig.json:
```
V™Ö,Ô=5!‹kÉô–õA÷§SÔùÈ¨!ÒÑuxR×B’Üµs<<£¼)[¤XN¢ž1½[Áuóg¯h¾ÜZÔ¿&'„,“t^Ìï‰„¿Cvžð£P¸… Sî€·Ðy™¾ìªÝ&DžB…
÷åzx9tœEr#åìiå§À ?´Ã(<šPß:.CqÅ’é¡DX0H7i4ÂÐ³…>
 +¢µ³Þ•§zI5oÀtIº+ktšáÒø…z0hÚ),ªÈVÇ·Iîá·JËê}”Þâ:¿4#»«ns+è˜òÞgÔ¿ÐJ×³ºò0—d» \ zGù´¥)ÌŽl¸YÝTp¾êxE¹OHß¨éìÏÿhÒ2÷Àß"¬¸zñ_ƒ5Ù\¤×Ñø’tÁ’ôŒ¦!ŽõìtìÍ»KB
Ìñ“cZÁquoÁ
y¯!óçE‹öÔÞ)Â_÷ºÖJLÐcÒ~
ÒfÀ¢"ÎL!Iæ‰nÓLFŒÐþeÕ“J‰Ÿf+
Eý‰,J8;¤uˆh„[¢•õäÃ[@ëåÆ‘xG]…K¸«ë{Æ­¦é“ÚgÔÃ"Ç´ìôC‡3ëÏö¬¬YæcJØˆI”u˜D
çíFè= >ªœC£s^´+Ï]¶ÂUÔj¢?z0DizLÓóÅò=q ¥”lÕ
É× þÿ Ü67D_-|Õ­w’²ôë`¸ç$ÁgÊfÞÀçtœ“
ê’\:öÉ
›LK“0BdÖ™ìŸS¨ÐÒ(Ë¶ÿh+­±11íÓ„»„xÕÂã‰Ù±ÂJTšß¦º88É¡ykøÂLó\#kÀ­ÆlcáR(P<qï°ˆ²®5XJï¢þ{º×êolÒÄ.§çH¹2¦;S¤—³ºO¹4²èÄâ½P&
z8ë…Â'f±ÔÔÅ·äµÕC®WY;4wˆ0rÁØ¹KöÎ¿ä;4ŽS5ÀûÜ^½
```

Makes me wonder: could it be due to hard disk encryption, because that's enabled on my machine? On the other hand, other (text based) files are just fine in the same directory, like the bak files (not sure how they got in there, they are almost 2y old) or the logs in de logs subdirectory. Or maybe saving during shutdown causes this? Is saving during exit required, because settings are saved immediately after changing them?

I compared the files after a second boot this morning, forgot about it when I first booted up. Seeing the logs, the error appeared after the second boot. I was connected to a VPN when I restarted the computer.

A workaround could be to restore settings from the backup folder if the application is unable to read the "current" settings. Maybe have it as an option to do so.

Just thinking out load here and giving you as much information as possible.

Log file extract from yesterday till after second reboot:
```
2020-05-07 18:31:23.661 - ShareX closing.
2020-05-07 18:31:23.680 - ApplicationConfig save started: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-07 18:31:24.174 - ApplicationConfig save successful: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-07 18:31:24.174 - UploadersConfig save started: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-07 18:31:24.349 - UploadersConfig save successful: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-07 18:31:24.350 - HotkeysConfig save started: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-07 18:31:24.539 - HotkeysConfig save successful: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-07 18:31:24.540 - ShareX closed.
2020-05-08 08:38:52.053 - ShareX starting.
2020-05-08 08:38:52.075 - Version: 13.1
2020-05-08 08:38:52.113 - Build: Release
2020-05-08 08:38:52.113 - Command line: "C:\Program Files\ShareX\ShareX.exe" -silent
2020-05-08 08:38:52.113 - Personal path: C:\Users\<username>\Documents\ShareX
2020-05-08 08:38:52.128 - Operating system: Windows 10 Enterprise (64-bit)
2020-05-08 08:38:52.128 - Running as elevated process: False
2020-05-08 08:38:52.135 - Flags: SilentRun
2020-05-08 08:38:52.170 - ApplicationConfig load started: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-08 08:38:54.056 - ApplicationConfig load finished: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-08 08:38:54.076 - UploadersConfig load started: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-08 08:38:54.076 - MainForm init started.
2020-05-08 08:38:55.291 - UploadersConfig load finished: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-08 08:38:55.302 - HotkeysConfig load started: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-08 08:38:55.373 - HotkeysConfig load finished: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-08 08:38:55.645 - MainForm init finished.
2020-05-08 08:38:57.408 - Startup time: 5513 ms
2020-05-08 08:38:57.411 - CommandLine: silent
2020-05-08 08:38:57.462 - Hotkey registered: Hotkey: Ctrl + Print Screen, Description: Capture region, Job: RectangleRegion
2020-05-08 08:38:57.463 - Hotkey registered: Hotkey: Print Screen, Description: Capture entire screen, Job: PrintScreen
2020-05-08 08:38:57.463 - Hotkey registered: Hotkey: Alt + Print Screen, Description: Capture active window, Job: ActiveWindow
2020-05-08 08:38:57.463 - Hotkey registered: Hotkey: Shift + Print Screen, Description: Start/Stop screen recording using custom region, Job: ScreenRecorder
2020-05-08 08:38:57.463 - Hotkey registered: Hotkey: Ctrl + Shift + Print Screen, Description: Start/Stop screen recording (GIF) using custom region, Job: ScreenRecorderGIF
2020-05-08 08:38:57.466 - HotkeyManager started.
2020-05-08 08:38:57.468 - WatchFolderManager started.
2020-05-08 09:13:29.110 - ShareX closing.
2020-05-08 09:13:29.135 - ApplicationConfig save started: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-08 09:13:30.105 - ApplicationConfig save successful: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-08 09:13:30.107 - UploadersConfig save started: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-08 09:13:30.365 - UploadersConfig save successful: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-08 09:13:30.366 - HotkeysConfig save started: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-08 09:13:30.652 - HotkeysConfig save successful: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-08 09:13:30.653 - ShareX closed.
2020-05-08 09:16:09.318 - ShareX starting.
2020-05-08 09:16:09.358 - Version: 13.1
2020-05-08 09:16:09.377 - Build: Release
2020-05-08 09:16:09.391 - Command line: "C:\Program Files\ShareX\ShareX.exe" -silent
2020-05-08 09:16:09.391 - Personal path: C:\Users\<username>\Documents\ShareX
2020-05-08 09:16:09.392 - Operating system: Windows 10 Enterprise (64-bit)
2020-05-08 09:16:09.393 - Running as elevated process: False
2020-05-08 09:16:09.395 - Flags: SilentRun
2020-05-08 09:16:09.518 - ApplicationConfig load started: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json
2020-05-08 09:16:09.991 - ApplicationConfig load failed: C:\Users\<username>\Documents\ShareX\ApplicationConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: V. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:10.023 - ApplicationConfig load started: C:\Users\<username>\Documents\ShareX\Backup\ApplicationConfig.json
2020-05-08 09:16:10.143 - ApplicationConfig load failed: C:\Users\<username>\Documents\ShareX\Backup\ApplicationConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: V. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:10.144 - Loading new ApplicationConfig instance.
2020-05-08 09:16:10.341 - MainForm init started.
2020-05-08 09:16:10.744 - UploadersConfig load started: C:\Users\<username>\Documents\ShareX\UploadersConfig.json
2020-05-08 09:16:10.946 - UploadersConfig load failed: C:\Users\<username>\Documents\ShareX\UploadersConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: �. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:11.794 - MainForm init finished.
2020-05-08 09:16:11.937 - Startup time: 3389 ms
2020-05-08 09:16:11.940 - CommandLine: silent
2020-05-08 09:16:12.183 - UploadersConfig load started: C:\Users\<username>\Documents\ShareX\Backup\UploadersConfig.json
2020-05-08 09:16:12.208 - UploadersConfig load failed: C:\Users\<username>\Documents\ShareX\Backup\UploadersConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: �. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:12.208 - Loading new UploadersConfig instance.
2020-05-08 09:16:12.389 - HotkeysConfig load started: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json
2020-05-08 09:16:12.465 - HotkeysConfig load failed: C:\Users\<username>\Documents\ShareX\HotkeysConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: �. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:12.537 - HotkeysConfig load started: C:\Users\<username>\Documents\ShareX\Backup\HotkeysConfig.json
2020-05-08 09:16:12.657 - HotkeysConfig load failed: C:\Users\<username>\Documents\ShareX\Backup\HotkeysConfig.json:
Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: �. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at ShareX.HelpersLib.SettingsBase`1.LoadInternal(String filePath, String backupFolder)
2020-05-08 09:16:12.657 - Loading new HotkeysConfig instance.
2020-05-08 09:16:12.677 - Hotkey registered: Hotkey: Ctrl + Print Screen, Description: Capture region, Job: RectangleRegion
2020-05-08 09:16:12.677 - Hotkey registered: Hotkey: Print Screen, Description: Capture entire screen, Job: PrintScreen
2020-05-08 09:16:12.677 - Hotkey registered: Hotkey: Alt + Print Screen, Description: Capture active window, Job: ActiveWindow
2020-05-08 09:16:12.677 - Hotkey registered: Hotkey: Shift + Print Screen, Description: Start/Stop screen recording using custom region, Job: ScreenRecorder
2020-05-08 09:16:12.677 - Hotkey registered: Hotkey: Ctrl + Shift + Print Screen, Description: Start/Stop screen recording (GIF) using custom region, Job: ScreenRecorderGIF
2020-05-08 09:16:12.680 - HotkeyManager started.
2020-05-08 09:16:12.683 - WatchFolderManager started.
```

**Comment by Jaex** [comment]
I installed Windows 10 virtual machine to test corruption issue by turning off vm (electricity loss or bsod simulation) or shutting it down (pressing power button in your pc case). And I could reproduce setting corruption in latest ShareX stable build. Later I downloaded dev build which contains this fix: https://github.com/ShareX/ShareX/commit/c63c37fd192d0bae651e580ed8105875111ba3a6 and I could not corrupt settings anymore. And I was sure that I could mimic same situation because ShareX logs were getting corrupted, yet settings were intact.

This fix involves bypassing os cache while writing settings to hard drive. You can read more info about os cache here: https://superuser.com/a/866732/223867

But there is a lot of reasons that peoples settings could be not saving at all too, for example if anti virus software is blocking ShareX access to Documents folder. I just hopefully fixed corrupting settings on shutdown/electricity lose issue.

**Comment by davidgilbertson** [comment]
Just FYI I'm a new user and I set a bunch of settings then a bit later noticed they were all gone (like shortcuts and audio device to use for recording). I've been playing around a lot and changed many things and rebooted so I don't know what caused it. Maybe that I changed the directory where screenshots are saved. I didn't realise that this wasn't just where screenshots were saved, but also where all the settings were saved, and I guess they aren't copied when I change the directory?

Anyway, just letting you know that saving settings is still not very robust (for a new user who doesn't know what they're doing).
